### PR TITLE
Adds distribution with built package

### PR DESCRIPTION
**Pull request by flovilmart**
- Adds gulp build system to build the parse-dashboard dist
- Adds instruction for safe production deployments
- Adds ability to build npm package
- npm package only contains dist/ and optimized bundles
- 5 steps configuration for production deployments.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -104,6 +104,16 @@ If you are not familiar with Docker, ``--port 8080`` with be passed in as argume
 
 ## Deploying in production
 
+For production deployments, it's recommended to use the npm pacakge
```
Comment (peterdotjs) [inline README.md]: package (sp)

File: package.json [hunk 2]
```diff
@@ -1,5 +1,15 @@
 {
-  "private": true,
+  "name": "parse-dashboard",
+  "description": "The parse dashboard",
+  "version": "1.0.0",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/ParsePlatform/parse-dashboard"
+  },
+  "license": "BSD-3-Clause",
```
Comment (peterdotjs) [inline package.json]: We'll need to reference the `license` file. 

`"license: "LicenseRef-LICENSE"` 

Ref: https://github.com/npm/npm/issues/8795

Comment (flovilmart) [inline package.json]: ok!

Comment (flovilmart) [inline package.json]: Put the same license as is parse-server

**Comment by peterdotjs** [comment]
other than the two comments, lgtm!

**Comment by flovilmart** [comment]
@peterdotjs version 1.0.0 is right? @drew-gross has the access to npmjs to publish the package, I can't publish it :(

**Comment by peterdotjs** [comment]
Only @drew-gross I think has publish perms. Drew could you give us permissions?

Hmm regarding version # we could go the same route as `parse-server` and start with 2.0.0? Thoughts @gfosco ?

**Comment by flovilmart** [comment]
bumped to 2.0.0, at publish time, @drew-gross could lower leave it

**Comment by drew-gross** [comment]
@flovilmart what benefit do we get from gulp in this case? It doesn't seem like it's doing much right now, and I don't really want to add an entire other build step unless it solves a really big problem.

**Comment by flovilmart** [comment]
@drew-gross webpack is really bad at moving files around, and it would likely require another package, like webpack-copy-files, a new webpack config, based on production. I don't mind removing gulp, but I tried with webpack and never got a good reliable solution. I'm open to remove gulp.

**Comment by drew-gross** [comment]
@flovilmart why would webpack need to move files around? Seems like you should be able to just change the output directory.

**Comment by flovilmart** [comment]
We need to copy the dashboard files into dist, that is not committed, and only created upon publish.

**Pull request merged**